### ISSUE TITLE
DLL not working on OBS 25.0.8/windows

### ISSUE THREAD

**Issue by gzalo** [fzwoch/obs-gstreamer#25]
Hi.
I'm having issues running the plugin under the latest version of OBS (25.0.8, 64-bit, windows)
I've tried both the release found here as well as one built from the latest commit here: https://gitlab.com/fzwoch/obs-gstreamer/pipelines

I have tried with the gstreamer 1.16.2 and also the 1.15.1 runtime, in both cases I've added manually the folder which contains the dlls (`D:\gstreamer\1.0\x86_64\bin`) to my path env var, and validated that gst-inspect-1.0.exe works just fine.

The only thing related to gstreamer that appears in the logs is:
`Module '../../obs-plugins/64bit/obs-gstreamer.dll' not loaded`

Many thanks, Gonzalo

**Comment by fzwoch** [comment]
Sounds like some DLL files cannot be found. Check with the dependencywalker.com tool if it reports something obvious.

Maybe it depends on what kind of gstreamer release you pick - msvc or mingw. The binary is linked to the mingw version. Msvc tends to strip the "lib" prefix from library names - so that may be something..

**Comment by gzalo** [comment]
Thanks! I was able to fix it. Dependency walker was not useful, so I switched to Process Monitor (a tool that lists every Windows API call) and detected that after loading that library, some of the additional dlls it loads were being pulled from another directory (in my case, the mingw directory), which contained a library with the same name but probably an older version.
So I changed the Mingw path to the bottom of the Path env var lists. Now it seems to be working just fine :D !

**Comment by mk0x9** [comment]
Had the same issue, added bin directory of mingw gstreamer to the system PATH

![image](https://user-images.githubusercontent.com/387358/88087704-a062a980-cb89-11ea-97df-a65add5e2279.png)

**Comment by philhartung** [comment]
I had a similar issue. Process Monitor solved it for me. There was Tesseract-OCR in my path, which includes some gstreamer dlls which the plugin tried to load, but I guess where incompatible. Removed Tesseract-OCR from my path to get it working. I have OBS v26.0.2, v0.2.0 of this plugin and gstreamer v1.16.1 (mingw)

**Comment by some1chan** [comment]
For anyone running into a similar issue, I found https://github.com/lucasg/Dependencies/ to be significantly more useful than http://dependencywalker.com/, as it loads in the `obs-gstreamer.dll` file instantly, compared to `depends.exe` not responding for >10 minutes, and then later giving out info I couldn't read. I'd recommend using this program over `depends.exe` or Process Monitor first. 

After following the red herring of `obs.dll` not being found, my real issue was this: `libglib-2.0-0.dll` was being fetched from the System32 folder, and is incompatible with `obs-gstreamer`.

![libglib-2.0-0.dll highlighted, with its path directed at System32](https://user-images.githubusercontent.com/31833556/218366746-899b1b94-fbaa-4423-afe2-0090c1b1f949.png "DependenciesGui.exe output of obs-gstreamer.dll")

According to [this post on the Qt forums](https://forum.qt.io/post/142210), I needed `obs-gstreamer.dll` to be either:

* The directory where the executable module for the current process is located
* The current directory

...before it reads from the Windows system directory, and causes the issue.

From there, I could solve the problem by copying `libglib-2.0-0.dll` in the plugins folder, or where `obs.exe` was. Logs are cleaner if I used `C:\Program Files\obs-studio\bin\64bit`, so I went with that, and it worked!
